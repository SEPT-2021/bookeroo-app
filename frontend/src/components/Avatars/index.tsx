import React from "react";
import { makeStyles, createStyles, Theme } from "@material-ui/core/styles";
import Avatar from "@material-ui/core/Avatar";
import person1 from "../../assets/Profile/p1.jpg";
import person2 from "../../assets/Profile/p2.jpg";
import person3 from "../../assets/Profile/p3.jpg";
import person4 from "../../assets/Profile/p4.jpg";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      display: "flex",
      "& > *": {
        margin: theme.spacing(1),
      },
    },
    small: {
      width: theme.spacing(3),
      height: theme.spacing(3),
    },
    large: {
      width: theme.spacing(7),
      height: theme.spacing(7),
    },
  })
);

export default function ImageAvatars() {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Avatar alt="P1" src={person1} className={classes.large} />
      <Avatar alt="P2" src={person2} className={classes.large} />
      <Avatar alt="P3" src={person3} className={classes.large} />
      <Avatar alt="P4" src={person4} className={classes.large} />
    </div>
  );
}
