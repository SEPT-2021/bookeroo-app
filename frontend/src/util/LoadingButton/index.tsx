import React from "react";
import { Button, ButtonProps, CircularProgress } from "@material-ui/core";

function LoadingButton({ loading, ...rest }: LoadingButtonProps) {
  return (
    <Button disabled={loading} {...rest}>
      {loading ? (
        <CircularProgress size={24} color="secondary" />
      ) : (
        rest.children
      )}
    </Button>
  );
}

interface LoadingButtonProps extends ButtonProps {
  loading?: boolean;
}

export default LoadingButton;
