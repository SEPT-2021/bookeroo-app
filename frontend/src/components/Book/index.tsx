import React from "react";
import { Button } from "@material-ui/core";
import AddIcon from "@material-ui/icons/Add";
import styled from "styled-components";
import { BookItemType } from "../../util/types";

export const Wrapper = styled.div`
  margin-top: 80px;
`;

type Props = {
  item: BookItemType;
  handleAddToCart: (clickedItem: BookItemType) => void;
};

const Book: React.FC<Props> = ({ item, handleAddToCart }) => (
  <Wrapper>
    <img src={item.cover} alt={item.title} />
    <div>
      <h1>{item.title}</h1>
      <h2> by {item.author}</h2>
      <p>Page Count : {item.pageCount}</p>
      <p>Book isbn : {item.isbn}</p>
      <p>Book description : {item.description}</p>
    </div>
    <Button onClick={() => handleAddToCart(item)}>
      <AddIcon />
      Add to cart
    </Button>
  </Wrapper>
);

export default Book;
