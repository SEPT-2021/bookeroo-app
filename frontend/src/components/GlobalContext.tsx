import React, {
  createContext,
  Dispatch,
  FC,
  SetStateAction,
  useCallback,
  useEffect,
  useState,
} from "react";
import { useQuery, useQueryClient } from "react-query";
import { api, profile } from "../util/api";
import { BookItemType, Listing, TokenProps, User } from "../util/types";
import useStickyState from "../util/useStickyState";

export interface CartItem {
  listing: Listing;
  book: BookItemType;
}

interface GlobalContextType {
  user?: User;

  login(data: TokenProps): void;

  signOut(): void;

  cartOpen: boolean;
  setCartOpen: Dispatch<SetStateAction<boolean>>;
  clearCart(): void;
  cartItems: CartItem[];

  addToCart(b: CartItem): void;

  removeFromCart(id: string): void;
}

export const GlobalContext = createContext<GlobalContextType>({} as never);

export const GlobalContextProvider: FC<unknown> = ({ children }) => {
  const client = useQueryClient();
  const [token, setToken] = useState<string>();
  const [cartOpen, setCartOpen] = useState(false);
  const [cartItems, setCartItems] = useStickyState<CartItem[]>([], "cart");
  const addToCart = (clickedItem: CartItem) => {
    setCartItems((prev) => {
      // 1. Is the item already added in the cart?
      const isItemInCart = prev.find(
        (item) => item.listing.id === clickedItem.listing.id
      );

      if (isItemInCart) {
        return cartItems;
      }
      // First time the item is added
      return [...prev, clickedItem];
    });
  };
  const removeFromCart = (id: string) => {
    setCartItems((prev) => prev.filter((l) => l.listing.id !== id));
  };
  const {
    data: userData,
    refetch,
    remove,
    isError,
  } = useQuery("user", profile);
  // On token change, update the header and refetch user.
  useEffect(() => {
    if (token) {
      api.defaults.headers.Authorization = `Bearer ${token}`;
      refetch();
    } else {
      delete api.defaults.headers.Authorization;
      remove();
    }
  }, [refetch, remove, token]);
  // On startup, get token from local storage
  useEffect(() => {
    setToken(localStorage.getItem("token") || undefined);
  }, []);
  const signOut = useCallback(() => {
    localStorage.removeItem("token");
    setToken(undefined);
    client.invalidateQueries();
  }, [client]);
  const login = (data: TokenProps) => {
    localStorage.setItem("token", data.jwt as string);
    setToken(data.jwt);
    client.invalidateQueries();
  };
  // Clear token if error
  useEffect(() => {
    if (isError) {
      signOut();
    }
  }, [isError, signOut]);
  return (
    <GlobalContext.Provider
      value={{
        user: token ? userData : undefined,
        login,
        signOut,
        cartOpen,
        setCartOpen,
        clearCart() {
          setCartItems([]);
        },
        cartItems,
        addToCart,
        removeFromCart,
      }}
    >
      {children}
    </GlobalContext.Provider>
  );
};
