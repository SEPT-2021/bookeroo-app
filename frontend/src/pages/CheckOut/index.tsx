import React, { useState } from "react";
import { createStyles, Theme } from "@material-ui/core/styles";
import { withStyles, WithStyles } from "@material-ui/core";
import { useMutation } from "react-query";
import FormField from "../../util/FormField";
import LoadingButton from "../../util/LoadingButton";
import { checkout } from "../../util/api";
import { CartType, DataItemType } from "../../util/types";

function CheckOut({ classes }: CheckOutProps) {
  const [addressLine1, setAddressLine1] = useState("");
  const [addressLine2, setAddressLine2] = useState("");
  const [city, setCity] = useState("");
  const [state, setState] = useState("");
  const [postalCode, setPostalCode] = useState("");
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const [data, setData] = useState<CartType>();

  const {
    data: checkoutData,
    mutate,
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    error,
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    isLoading,
    reset,
  } = useMutation(checkout);
  const getLocal = () => {
    return JSON.parse(localStorage.getItem("cart") as string) as DataItemType[];
  };

  const getShippingAddress = () => ({
    addressLine1,
    addressLine2,
    city,
    state,
    postalCode,
  });

  const makeData: () => CartType = () => {
    // eslint-disable-next-line @typescript-eslint/no-shadow
    return {
      orderItems: getLocal(),
      shippingAddress: getShippingAddress(),
    };
  };

  const onSubmit = () => {
    const newData = makeData();
    setData(newData);
    mutate(newData);
    // eslint-disable-next-line no-console
    console.log(newData);
  };
  if (checkoutData) {
    window.open(checkoutData);

    reset();
  }
  return (
    <div className={classes.root}>
      <form className={classes.root} noValidate autoComplete="off">
        <div>
          <FormField
            required
            name="addressLine1"
            label="Address Line 1"
            placeholder="Address Line 1"
            onChange={setAddressLine1}
            multiline
          />
          <FormField
            name="addressLine2"
            label="Address Line 2"
            placeholder="Address Line 2"
            onChange={setAddressLine2}
            multiline
          />
          <FormField
            required
            name="city"
            label="City"
            placeholder="City"
            onChange={setCity}
            multiline
          />
          <FormField
            required
            name="state"
            label="State"
            placeholder="State"
            onChange={setState}
            multiline
          />
          <FormField
            required
            name="postalCode"
            label="Postal Code"
            placeholder="Postal Code"
            onChange={setPostalCode}
            multiline
          />
          <div>
            <LoadingButton
              variant="contained"
              color="primary"
              onClick={onSubmit}
            >
              Submit
            </LoadingButton>
          </div>
        </div>
      </form>
    </div>
  );
}

const styles = (theme: Theme) => {
  return createStyles({
    root: {
      "& .MuiTextField-root": {
        margin: theme.spacing(1),
        width: "25ch",
        gridArea: "center",
        marginTop: "100px",
      },
    },
    button: {},
  });
};

type CheckOutProps = WithStyles<typeof styles>;

export default withStyles(styles)(CheckOut);
