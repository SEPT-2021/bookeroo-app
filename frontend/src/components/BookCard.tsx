import React, { useContext } from "react";
import { makeStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import CardMedia from "@material-ui/core/CardMedia";
import Typography from "@material-ui/core/Typography";
import {
  Avatar,
  CardActions,
  CardHeader,
  CircularProgress,
  Collapse,
  IconButton,
} from "@material-ui/core";
import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
import { AddShoppingCart, Book, Delete, Launch } from "@material-ui/icons";
import { useMutation, useQueryClient } from "react-query";
import { CardActionArea } from "@mui/material";
import { useHistory } from "react-router-dom";
import { BookItemType, Role } from "../util/types";
import { GlobalContext } from "./GlobalContext";
import { deleteBookById } from "../util/api";
import Link from "../util/Link";

const useStyles = makeStyles((theme) => ({
  card: {
    margin: theme.spacing(2),
  },
  media: {
    height: 250,
  },
}));

type BookProps = {
  book: BookItemType;
  checked: boolean;
  onCartClick: () => void;
};

export default function BookCard({
  book: { cover: imageUrl, id, title, description, author, pageCount, price },
  checked,
  onCartClick,
}: BookProps) {
  const classes = useStyles();
  const link = `/book/${id}`;
  const history = useHistory();
  const [expanded, setExpanded] = React.useState(false);
  const { user } = useContext(GlobalContext);
  const client = useQueryClient();
  const { mutate: deleteMutate, isLoading } = useMutation(deleteBookById, {
    onSuccess: () => {
      client.invalidateQueries("books");
    },
  });
  const handleExpandClick = () => {
    setExpanded(!expanded);
  };

  return (
    <Collapse in={checked} timeout={checked ? 1500 : undefined}>
      <Card className={classes.card}>
        <CardHeader
          avatar={
            <Avatar aria-label="recipe">
              <Book />
            </Avatar>
          }
          action={
            user?.role === Role.ROLE_ADMIN && (
              <IconButton onClick={() => deleteMutate({ id: String(id) })}>
                {isLoading ? (
                  <CircularProgress size={24} color="secondary" />
                ) : (
                  <Delete />
                )}
              </IconButton>
            )
          }
          title={<b>{title}</b>}
          subheader={author}
        />
        <CardActionArea onClick={() => history.push(link)}>
          <CardMedia className={classes.media} image={imageUrl} title={title} />
          <CardContent>
            <Typography variant="body2" color="textSecondary" component="p">
              Page Count: {pageCount}
              <br />
              Price: {price}
            </Typography>
          </CardContent>
        </CardActionArea>
        <CardActions disableSpacing>
          <IconButton aria-label="add to favorites" onClick={onCartClick}>
            <AddShoppingCart />
          </IconButton>
          <Link to={link}>
            <IconButton aria-label="share">
              <Launch />
            </IconButton>
          </Link>
          <IconButton
            onClick={handleExpandClick}
            aria-expanded={expanded}
            aria-label="show more"
          >
            <ExpandMoreIcon />
          </IconButton>
        </CardActions>
        <Collapse in={expanded} timeout="auto" unmountOnExit>
          <CardContent>
            <Typography paragraph>{description}</Typography>
          </CardContent>
        </Collapse>
      </Card>
    </Collapse>
  );
}
