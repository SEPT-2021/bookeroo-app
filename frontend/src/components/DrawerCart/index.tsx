import React, { useContext } from "react";
import { Badge, Drawer, Grid, IconButton } from "@material-ui/core";
import styled from "styled-components";
import { ShoppingCartOutlined } from "@material-ui/icons";
import CartItem from "../CartItem/CartItem";
import Link from "../../util/Link";
import { BookItemType } from "../../util/types";
import { GlobalContext } from "../GlobalContext";

const Wrapper = styled.aside`
  font-family: Arial, Helvetica, sans-serif;
  width: 500px;
  padding: 20px;
`;

type Props = {
  cartItems: BookItemType[];
  addToCart: (clickedItem: BookItemType) => void;
  removeFromCart: (id: number) => void;
};

const Cart: React.FC<Props> = ({ cartItems, addToCart, removeFromCart }) => {
  const calculateTotal = (items: BookItemType[]) =>
    items.reduce((ack: number, item) => ack + item.amount * item.price, 0);

  return (
    <Wrapper>
      <h2>Your Shopping Cart</h2>
      {cartItems.length === 0 ? <p>No items in cart.</p> : null}
      {cartItems.map((item) => (
        <CartItem
          key={item.id}
          item={item}
          addToCart={addToCart}
          removeFromCart={removeFromCart}
        />
      ))}
      <h2>Total: ${calculateTotal(cartItems).toFixed(2)}</h2>
      <Grid item>
        <Link to="/checkout" variant="body2">
          Checkout
        </Link>
      </Grid>
    </Wrapper>
  );
};

const StyledButton = styled(IconButton)`
  position: fixed;
  right: 20px;
  top: 90px;
  background-color: orange;
`;

export default function DrawerCart() {
  const { addToCart, removeFromCart, cartItems, setCartOpen, cartOpen } =
    useContext(GlobalContext);

  const getTotalItems = (items: BookItemType[]) =>
    items.reduce((ack: number, item) => ack + item.amount, 0);
  return (
    <>
      <StyledButton onClick={() => setCartOpen(true)}>
        <Badge badgeContent={getTotalItems(cartItems)} color="error">
          <ShoppingCartOutlined />
        </Badge>
      </StyledButton>
      <Drawer anchor="right" open={cartOpen} onClose={() => setCartOpen(false)}>
        <Cart
          cartItems={cartItems}
          addToCart={addToCart}
          removeFromCart={removeFromCart}
        />
      </Drawer>
    </>
  );
}
