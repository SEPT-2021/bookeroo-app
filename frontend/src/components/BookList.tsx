import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import { Grid } from "@material-ui/core";
import BookCard from "./BookCard";
import useWindowPosition from "../hook/useWindowPosition";
import { BookItemType } from "../util/types";

const useStyles = makeStyles((theme) => ({
  root: {
    minHeight: "100vh",
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    [theme.breakpoints.down("md")]: {
      flexDirection: "column",
    },
  },
  gridItem: {
    width: "100%",
  },
}));

export default function BookList({
  books,
  checked: argChecked = false,
}: {
  books: BookItemType[];
  checked?: boolean;
}) {
  const classes = useStyles();
  const hookChecked = useWindowPosition("header");
  const checked = argChecked || hookChecked;

  return (
    <div className={classes.root} id="book-list">
      <Grid container id="header" alignItems="center" justifyContent="center">
        {books.map((book) => (
          <Grid
            item
            md={3}
            sm={6}
            xs={12}
            className={classes.gridItem}
            key={book.id}
          >
            <BookCard book={book} key={book.title} checked={checked} />
          </Grid>
        ))}
      </Grid>
    </div>
  );
}
