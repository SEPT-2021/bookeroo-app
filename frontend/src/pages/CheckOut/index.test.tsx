import React from "react";
import ReactDOM from "react-dom";
import { cleanup, render } from "@testing-library/react";
import { QueryClient, QueryClientProvider } from "react-query";
import { BrowserRouter } from "react-router-dom";
import CheckOut from "./index";
import LoadingButton from "../../util/LoadingButton";
import { GlobalContextProvider } from "../../components/GlobalContext";

afterEach(cleanup);
describe("Rendering page CheckOut", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <QueryClientProvider client={new QueryClient()}>
          <GlobalContextProvider>
            <CheckOut />
          </GlobalContextProvider>
        </QueryClientProvider>
      </BrowserRouter>,
      div
    );
    expect(div).toMatchSnapshot();
  });
});

describe("LoadingButton render matches snapshot", () => {
  test("renders button", () => {
    const button = render(<LoadingButton />);
    expect(button).toMatchSnapshot();
  });
});
