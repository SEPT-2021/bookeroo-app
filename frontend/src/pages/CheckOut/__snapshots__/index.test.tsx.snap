// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingButton render matches snapshot renders button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Rendering page CheckOut renders without crashing 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined css-11lq3yg-MuiGrid-root"
    style="margin-top: 120px;"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6 css-tn5ix5-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-1jc04hb"
      >
        <main
          class="MuiContainer-root MuiContainer-maxWidthSm css-nicjba-MuiContainer-root"
        >
          <div
            class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded css-126nd9x-MuiPaper-root"
          >
            <h1
              class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter css-149xwtl-MuiTypography-root"
            >
              Checkout
            </h1>
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
            >
              Shipping address
            </h6>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-wdtwov-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                    data-shrink="false"
                    for="firstName"
                    id="firstName-label"
                  >
                    First name
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="given-name"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      id="firstName"
                      name="firstName"
                      required=""
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-3"
                      >
                        <span>
                          First name
                           *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-wdtwov-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                    data-shrink="false"
                    for="lastName"
                    id="lastName-label"
                  >
                    Last name
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="family-name"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      id="lastName"
                      name="lastName"
                      required=""
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-3"
                      >
                        <span>
                          Last name
                           *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1em9ut6-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                    data-shrink="false"
                    for="address1"
                    id="address1-label"
                  >
                    Address line 1
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="shipping address-line1"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      id="address1"
                      name="shippingAddress.addressLine1"
                      required=""
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-3"
                      >
                        <span>
                          Address line 1
                           *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1em9ut6-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                    data-shrink="false"
                    for="address2"
                    id="address2-label"
                  >
                    Address line 2
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="shipping address-line2"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      id="address2"
                      name="shippingAddress.addressLine2"
                      required=""
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-3"
                      >
                        <span>
                          Address line 2
                           *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-wdtwov-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                    data-shrink="false"
                  >
                    City
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="shipping address-level2"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      name="shippingAddress.city"
                      required=""
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-3"
                      >
                        <span>
                          City
                           *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-wdtwov-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                    data-shrink="false"
                    for="state"
                    id="state-label"
                  >
                    State/Province/Region
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      id="state"
                      name="shippingAddress.state"
                      required=""
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-3"
                      >
                        <span>
                          State/Province/Region
                           *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-wdtwov-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                    data-shrink="false"
                    for="zip"
                    id="zip-label"
                  >
                    Post Code
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      autocomplete="shipping postal-code"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      id="zip"
                      name="shippingAddress.postalCode"
                      required=""
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-3"
                      >
                        <span>
                          Post Code
                           *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1em9ut6-MuiGrid-root"
              >
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                >
                  <span
                    class="MuiCheckbox-root MuiCheckbox-colorSecondary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-root css-1sg8fd1-MuiButtonBase-root-MuiCheckbox-root"
                  >
                    <input
                      class="PrivateSwitchBase-input css-1m9pwf3"
                      data-indeterminate="false"
                      name="saveAddress"
                      type="checkbox"
                      value="yes"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckBoxOutlineBlankIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                  >
                    Use this address for payment details
                  </span>
                </label>
              </div>
            </div>
            <div
              class="MuiBox-root css-1bvc4cc"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained"
                style="margin-top: 45px; margin-left: 15px;"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Place Order With Paypal
                </span>
              </button>
            </div>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1vl95ex-MuiTypography-root"
          >
            Copyright © 
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorInherit"
              href="/"
            >
              Bookeroo
            </a>
             
            2021
            .
          </p>
        </main>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6 css-tn5ix5-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-1p4tv2l"
      >
        <h2>
          Your Shopping Cart
        </h2>
        <p>
          No items in cart.
        </p>
        <h2>
          Total: $
          0.00
        </h2>
      </div>
    </div>
  </div>
</div>
`;
